### ISSUE TITLE
tensorflow loss=NaN while training

### ISSUE THREAD

**Issue by rutvi1462** [keras-team/keras#14524]
Hello everyone, I am new in AI and I trying to learn [object detection using tensorflow 2](https://gilberttanner.com/blog/tensorflow-object-detection-with-tensorflow-2-creating-a-custom-model).

But I have an issue while train a model. 
Whenever I tried to run following command in cmd:

> python model_main_tf2.py \
>     --pipeline_config_path=training/ssd_efficientdet_d0_512x512_coco17_tpu-8.config \
>     --model_dir=training \
>     --alsologtostderr

I got following output:

> WARNING:tensorflow:Gradients do not exist for variables ['top_bn/gamma:0', 'top_bn/beta:0'] when minimizing the loss.
> W0313 11:59:25.249265 11204 utils.py:81] Gradients do not exist for variables ['top_bn/gamma:0', 'top_bn/beta:0'] when minimizing the loss.
> INFO:tensorflow:Step 4100 per-step time 2.775s loss=nan
> I0313 12:02:55.914595 15108 model_lib_v2.py:665] Step 4100 per-step time 2.775s loss=nan
> INFO:tensorflow:Step 4200 per-step time 2.202s loss=nan
> I0313 12:06:41.231918 15108 model_lib_v2.py:665] Step 4200 per-step time 2.202s loss=nan
> INFO:tensorflow:Step 4300 per-step time 1.964s loss=nan
> I0313 12:10:40.373922 15108 model_lib_v2.py:665] Step 4300 per-step time 1.964s loss=nan

Here you can observe that I got **_loss=nan_**. But in the tutorial output is different of the above command.

**Here is my tensorboard view:**

![Screenshot (222)](https://user-images.githubusercontent.com/63718746/111022119-e0b3b800-83f6-11eb-9c60-9735028e6869.png)
![Screenshot (223)](https://user-images.githubusercontent.com/63718746/111022126-ee693d80-83f6-11eb-8743-01053f915def.png)

I thought that because of **_loss=nan_** my model does not detect object.
Anyone has any solution for this?

**Comment by ymodak** [comment]
You may want to try recently updated Object Detection tutorials https://github.com/tensorflow/models/tree/master/research/object_detection/colab_tutorials
